### PR TITLE
Update Qase SDK version and dependencies

### PR DESCRIPTION
**Pull request by NikAzanov**
- Use updated `qaseio` sdk (^4.0.0)
- Drop `Python 3.10` support
- Drop `pytest-variables` support (now use pytest options to provide browser name)
- Small code and structure improvements
- Configure `ruff` to use as linter/formatter
- Configure `cspell` to fix typos

### CODE REVIEW

File: pytest_qaseio/plugin.py [hunk 1]
```diff
@@ -92,9 +79,9 @@ def pytest_configure(config: pytest.Config):
     qase_enabled = config.getoption("--qase-enabled")
     if not qase_enabled:
         return
-    browser = config.getoption("--webdriver")
+    browser: str = config.getoption("--webdriver")
     if browser == "remote":
-        browser = _get_browser_name(config, default=browser)
+        browser = config.getoption("--remote-browser", default="chrome")
```
Comment (M1troll) [inline pytest_qaseio/plugin.py:84]: Maybe store webdriver name only in the `--webdriver` argument, and the fact that we are using a remote browser should be specified via some bool argument? (e.g. `--webdriver-remote`)

File: pytest_qaseio/plugin.py [hunk 2]
```diff
@@ -189,7 +176,7 @@ def pytest_collection_modifyitems(
                 pytest.exit(e.message)
 
     @pytest.hookimpl(tryfirst=True, hookwrapper=True)
-    def pytest_runtest_makereport(self, item: pytest.Function):
+    def pytest_runtest_makereport(self, item: pytest.Function):  # noqa: ANN201
```
Comment (M1troll) [inline pytest_qaseio/plugin.py:179]: Can you use `Iterator[None]` ?

File: pytest_qaseio/plugin.py (hunk 1 above)
Comment (ron8mcr) [reply pytest_qaseio/plugin.py:84]: @M1troll suggested good option, but I think passing of browser name should be reworked to use custom hook like `pytest_qase_browser_name` (similar to `pytest_qase_file_storages`).
And default implementation of this hook may use `--webdriver` or `--remote-browser` options

File: pyproject.toml [hunk 3]
```diff
@@ -76,45 +76,166 @@ warn_redundant_casts = true
...
-include_trailing_comma = true
+[[tool.mypy.overrides]]
+module = "qaseio.*"
+implicit_reexport = true
+
+
+[tool.ruff]
+line-length = 100
+indent-width = 4
+target-version = "py310"
```
Comment (ron8mcr) [inline pyproject.toml:87]: 3.10 or 3.11?

File: pyproject.toml [hunk 4]
```diff
@@ -76,45 +76,166 @@ warn_redundant_casts = true
...
+exclude = [
+    "config",
+    ".venv",
+    "venv",
+    "__init__.py",
+    "plugins/selenium_plugin",
```
Comment (ron8mcr) [inline pyproject.toml:93]: There is no such dir in this repo

File: pytest_qaseio/debug_info.py [hunk 5]
```diff
@@ -42,19 +44,19 @@ def _extract_browser_log(self) -> str:
...
         timestamp_format = "%Y-%m-%d %H:%M:%S.%f"
-        entries = []
+        entries: list[str] = []
         for entry in log:
             timestamp = arrow.get(entry["timestamp"] / 1000.0).strftime(timestamp_format)
             entries.append(f"{timestamp} {entry['level']} - {entry['message']}")
-        log = "\n".join(entries)
-        return log
+        str_log = "\n".join(entries)
+        return str_log
```
Comment (ron8mcr) [inline pytest_qaseio/debug_info.py:59]:
```suggestion
        return "\n".join(entries)
```

File: pytest_qaseio/storage.py [hunk 6]
```diff
@@ -1,15 +1,31 @@
 import io
-from typing import Protocol
+from typing import Protocol, cast
 
 import qaseio
-import qaseio.apis
+from qaseio.configuration import Configuration
```
Comment (ron8mcr) [inline pytest_qaseio/storage.py:5]:
```suggestion
from qaseio import configuration as qaseio_config
```
To be consistent with `api_client.py`

File: pytest_qaseio/storage.py [hunk 7]
```diff
@@ -1,15 +1,31 @@
...
+
+class FileIO(io.BytesIO):
+    """Represent file object to pass in Qase API methods.
+
+    Set `name` and `mime` attributes to objects since it's required in
+    https://github.com/qase-tms/qase-python/blob/44d19a500246017a30e0fa06b35cace065135d96/qaseio/src/qaseio/api_client.py#L518
+
+    """
+
+    mime = "application/octet-stream"
```
Comment (ron8mcr) [inline pytest_qaseio/storage.py:24]: It's better to set proper mime type for images, so browsers can use default image viewer

**Review by ron8mcr — APPROVED** [review]

**Pull request merged**